lib/vrl/compiler/src/function.rs

```diff
@@ -116,8 +116,8 @@ impl ArgumentList {
         self.optional_expr(keyword)
             .map(|expr| match expr {
                 Expr::Literal(literal) => Ok(literal),
-                Expr::Variable(var) if var.value().is_some() => {
-                    match var.value().unwrap().clone().into_expr() {
+                Expr::Variable(var) if var.as_value().is_some() => {
```

Comment:
Not sure I understand this.. By changing the call from `value()` to `as_value()` we are doubling the amount of clones here.

Could we keep `value()` so that we only call `as_value()` when we actually need to own the Value?

Reply:
Yes, there's an extra clone, but this happens at compile-time, and this reduces the surface area of the API.

I don't consider clones to be "evil", you just should avoid them in hot paths, which this is not.

Reply:
A good call. I missed the not in a hot path bit.